op-batcher/batcher/channel_manager.go

```diff
@@ -154,43 +161,49 @@ func (s *channelManager) nextTxData(channel *channel) (txData, error) {
 // full, it only returns the remaining frames of this channel until it got
 // successfully fully sent to L1. It returns io.EOF if there's no pending tx data.
 //
-// It will generate the tx data internally, and decide whether to switch DA type
-// automatically. When switching DA type, the channelManager state will be rebuilt
+// It will decide whether to switch DA type automatically.
+// When switching DA type, the channelManager state will be rebuilt
 // with a new ChannelConfig.
 func (s *channelManager) TxData(l1Head eth.BlockID) (txData, error) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
 	channel, err := s.getReadyChannel(l1Head)
 	if err != nil {
 		return emptyTxData, err
 	}
-	if channel != nil && !channel.NoneSubmitted() {
+	if channel == nil {
+		panic("nil channel and nil err returned from getReadyChannel")
+	}
```

Comment:
Why this explicit panic check? the next line will panic anyways if `channel` was `nil`. I also fail to see how `getReadyChannel` could return `nil, nil`.

Reply:
I agree, removed in [7b740f7](https://github.com/ethereum-optimism/optimism/pull/12002/commits/7b740f739d9b8bebd9e18d491184e231dadf5772)